airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/file/SpillFileProvider.kt

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.load.file
+
+import io.airbyte.cdk.load.command.DestinationConfiguration
+import io.micronaut.context.annotation.Secondary
+import jakarta.inject.Singleton
+import java.nio.file.Files
+import java.nio.file.Path
+
+interface SpillFileProvider {
+    fun createTempFile(): Path
+}
+
+@Singleton
+@Secondary
+class DefaultSpillFileProvider(val config: DestinationConfiguration) : SpillFileProvider {
+    override fun createTempFile(): Path {
+        val directory = config.tmpFileDirectory
+        Files.createDirectories(directory)
+        return Files.createTempFile(directory, "staged-raw-records", "jsonl")
```

Comment:
random thought, not for this PR: right now, if I run a test and kill it before it can clean up these files, I get random cruft in `airbyte-integrations/connectors/destination-s3-v2/airbyte-cdk-load/`

so it would be nice to either (a) have tests manually nuke those files, or (b) be able to configure that path to point at `/tmp/whatever` from outside the connector